### PR TITLE
no-non-null-assertation rule added to tseslint

### PR DESCRIPTION
**Pull request by roxtraAK**

### CODE REVIEW

File: datatable/readXlsxFromAttachment-test.roxtest.ts [hunk 1]
```diff
@@ -26,16 +26,18 @@ describe("services", () => {
...
-        const fieldValue = resultField!.value as IDataTableFieldValue;
-        expect(fieldValue).to.not.be.undefined;
-        expect(fieldValue.rows.length).to.equal(20);
-        for (const row of fieldValue.rows) {
-          expect(row.selected).to.equal(false);
-          expect(typeof row.data.Name).to.equal("string");
-          expect(typeof row.data.Adresse).to.equal("string");
-          expect(typeof row.data.Telefonnummer).to.equal("string");
-          expect(typeof row.data.Kostenstelle).to.equal("number");
+        if (resultField) {
```
Comment (ceisele-r) [inline datatable/readXlsxFromAttachment-test.roxtest.ts:29]: Why is this necessary? The non-null assertion should be allowed here as its a test file.

File: ics/generate-service.ts [hunk 2]
```diff
@@ -177,7 +177,7 @@ export async function generate(environment: IServiceTaskEnvironment): Promise<bo
     }
     // The property instance.extras.fieldContents[targetField] is definitely defined as it was checked or initialized some lines above
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-    instance.extras.fieldContents[targetField]!.value = [url];
+    instance.extras.fieldContents[targetField].value = [url];
```
Comment (ceisele-r) [inline ics/generate-service.ts:180]: Which rule required this to be removed?

File: mssql/executequery-service.ts [hunk 3]
```diff
@@ -75,7 +75,7 @@ export async function executeQuery(environment: IServiceTaskEnvironment, configP
       }
       // The property environment.instanceDetails.extras.fieldContents[targetField] is definitely assigned here as it is checked or initialized in the lines above
       // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-      environment.instanceDetails.extras.fieldContents[targetField]!.value = rows[0].result;
+      environment.instanceDetails.extras.fieldContents[targetField].value = rows[0].result;
```
Comment (ceisele-r) [inline mssql/executequery-service.ts:78]: Which rule required this to be removed?

File: mysql/executequery-service.ts [hunk 4]
```diff
@@ -112,7 +112,7 @@ export async function executeQuery(environment: IServiceTaskEnvironment, configP
       }
       // The property environment.instanceDetails.extras.fieldContents[targetField] is definitely assigned here as it is checked or initialized in the lines above
       // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-      environment.instanceDetails.extras.fieldContents[targetField]!.value = res[0]["result"];
+      environment.instanceDetails.extras.fieldContents[targetField].value = res[0]["result"];
```
Comment (ceisele-r) [inline mysql/executequery-service.ts:115]: Which rule required this to be removed?

File: report/createReport-service.ts [hunk 5]
```diff
@@ -27,7 +27,7 @@ export function initReportUploadField(url: string, instance: IInstanceDetails, r
     }
     // The property environment.instanceDetails.extras.fieldContents[targetField] is definitely assigned here as it is checked or initialized in the lines above
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-    instance.extras.fieldContents[reportFieldName]!.value = [url];
+    instance.extras.fieldContents[reportFieldName].value = [url];
```
Comment (ceisele-r) [inline report/createReport-service.ts:30]: Which rule required this to be removed?

**Review by ceisele-r — CHANGES_REQUESTED** [review]

File: ics/generate-service.ts (hunk 2 above)
Comment (roxtraAK) [reply ics/generate-service.ts:180]: no non null assertion

File: datatable/readXlsxFromAttachment-test.roxtest.ts (hunk 1 above)
Comment (roxtraAK) [reply datatable/readXlsxFromAttachment-test.roxtest.ts:29]: yes my bad, this change was made before i excluded roxtest files from the eslint config

File: mssql/executequery-service.ts (hunk 3 above)
Comment (roxtraAK) [reply mssql/executequery-service.ts:78]: same as above

**Review by ceisele-r — APPROVED** [review]
